tests/rptest/tests/cluster_quota_test.py

```diff
@@ -100,3 +102,120 @@ def test_partition_throttle_mechanism(self):
         assert foo_response[0]['ErrorCode'] == 0  # success
         assert baz_response[0]['ErrorCode'] == 89  # throttling_quota_exceeded
         assert response['ThrottleMillis'] > 0
+
+
+class ClusterRateQuotaTest(RedpandaTest):
+    """
+    Ducktape tests for rate quota
+    """
+    topics = (TopicSpec(), )
+
+    def __init__(self, *args, **kwargs):
+        self.max_throttle_time = 10
+        super().__init__(*args,
+                         num_brokers=3,
+                         extra_rp_conf={
+                             "max_kafka_throttle_delay_ms":
+                             self.max_throttle_time
+                         },
+                         resource_settings=ResourceSettings(num_cpus=1),
+                         **kwargs)
+
+    def produce(self, producer, amount):
+        response_futures = [
+            producer.send(self.topic, self.msg) for _ in range(amount)
+        ]
+        for f in response_futures:
+            f.get(1)
```

Comment:
i see this CI failure happening. It might be the case that using this future-based API means you have to handle some exceptions like NotLeaderForPartitionError or do something else to deal with this.

I'd recommend taking quick look at the the client's github page / docs to see if there is a known way to deal with that. Worst case scenario we could probably catch it here and retry manually.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/ducktape/tests/runner_client.py", line 135, in run
    data = self.run_test()
  File "/usr/local/lib/python3.10/dist-packages/ducktape/tests/runner_client.py", line 227, in run_test
    return self.test_context.function(self.test)
  File "/root/tests/rptest/services/cluster.py", line 35, in wrapped
    r = f(self, *args, **kwargs)
  File "/root/tests/rptest/tests/cluster_quota_test.py", line 163, in test_client_group_rate_throttle_mechanism
    self.produce(producer, 10)
  File "/root/tests/rptest/tests/cluster_quota_test.py", line 129, in produce
    f.get(1)
  File "/usr/local/lib/python3.10/dist-packages/kafka/producer/future.py", line 65, in get
    raise self.exception # pylint: disable-msg=raising-bad-type
kafka.errors.NotLeaderForPartitionError: [Error 6] NotLeaderForPartitionError
```